compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1549,7 +1549,7 @@ object Parsers {
         else
           if sourceVersion.isAtLeast(`3.1`) then
             deprecationWarning(DeprecatedWithOperator(), withOffset)
-          makeAndType(t, withType())
+          makeAndType(t, withType()).withSpan(Span(startOffset(t), in.lastOffset, startOffset(t)))
```

Comment:
Should this be
```suggestion
          atSpan(in.lastOffset)(makeAndType(t, withType()))
```
?

I'm not familiar with the parser and how spans are supposed to be assigned. Perhaps someone else from the dotty team can check this? @nicolasstucki ?

Reply:
This variant has the proper point set on it. That may be the extra info that is missing for scalajs.
```suggestion
          atSpan(in.lastOffset, withOffset) { makeAndType(t, withType()) }
```

Reply:
Thanks! I applied the suggestion.